### PR TITLE
Various bug fixes and add colab notebook for running training using colab GPU's

### PR DESCRIPTION
**Pull request by zparis99**
There are 2 bug fixes, 1 config change, and 1 addition to this pr.

For bug fixes:
- commit: fix bug with matplotlib leading to unbounded RAM usage: Currently the matplotlib plots are not closed which leads to unbounded RAM usage as the number of plots increase. This is not all though since the RAM is not actually returned to the process unless the plotting process is killed. See https://stackoverflow.com/questions/28516828/memory-leaks-using-matplotlib?rq=4 for more details.  This commit adds a fix to this by plotting in a new process.
image
You can see the difference in this image. The first spike corresponds to running training without this change and the second flat line corresponds to RAM usage with the change.
- commit: fix learning rate scheduler to match the actual steps per epoch. Previously was overestimating since it wasn't taking into account batch size: The learning rate scheduler was previously overestimating the number of steps per epoch since it just took in the number of samples and didn't divide by batch size.

Config change:
- commit: add arg parser arguments to make data loading more configurable: This changes some argparser arguments such as removing the sandbox flag and instead adding a flag to point towards the root directory of the dataset to make it more configurable. This was necessary for the colab notebook since the path to the dataset will not always be the same. Default behavior will be the same as before though.

Addition:
- add colab notebook which can be used to train model to allow for quicker experimentation using free Google GPU's: Adds a ipynb notebook which can be easily ported to colab for experiments using GPU's there.

### CODE REVIEW

**Review by leoniekerken — APPROVED** [review]
Great, thank you!
